routing/single_mwm_router.cpp

```diff
@@ -215,6 +197,46 @@ bool SingleMwmRouter::LoadIndex(MwmSet::MwmId const & mwmId, string const & coun
   }
 }
 
+void SingleMwmRouter::BuildRoute(MwmSet::MwmId const & mwmId, IndexGraphStarter & starter, vector<Joint::Id> const & joints, RouterDelegate const & delegate, Route & route) const
```

Comment:
BuildRoute function can fail, but caller will never know this, also DoCalculateRoute will return NoError while error had happened.